### ISSUE TITLE
Inconsistent ansible-lint-ignore skip behavior with same file

### ISSUE THREAD

**Issue by pwalkr** [ansible/ansible-lint#5090]
##### Summary

I want to ignore both `yaml[commas]` and `yaml[colons]` on a large yaml data file aligned for readability. `.ansible-lint-ignore` spec is inconsistently honoring two `skip` directives.

This looks like the same issue as https://github.com/ansible/ansible-lint/issues/4730

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint 26.4.0 using ansible-core:2.21.0 ansible-compat:26.3.0 ruamel-yaml:0.19.1 ruamel-yaml-clib:0.2.15
```

- ansible installation method: OS package
- ansible-lint installation method: OS package

##### STEPS TO REPRODUCE

Create sample input `example.yml` and `.ansible-lint-ignore` in a directory. `example.yml`:
```yaml
example:
  foo:   bar
  fizz: [buzz,  fizzbuzz]
```
`.ansible-lint-ignore` to skip both rules:
```yaml
example.yml yaml[commas] skip
example.yml yaml[colons] skip
``` 
Run ansible-lint (second directive is demoted but not skipped)
```
$ ansible-lint example.yml
WARNING  Listing 1 violation(s) marked as ignored, likely already known
yaml[colons]: Too many spaces after colon (warning) # ignored
example.yml:2


Passed: 0 failure(s), 1 warning(s) in 1 files processed of 1 encountered. Last profile that met the validation criteria was 'production'.
```
v2: reordering `.ansible-lint-ignore`: (commas second)
```yaml
example.yml yaml[colons] skip
example.yml yaml[commas] skip
```
Output (now warns about commas instead)
```console
$ ansible-lint example.yml
WARNING  Listing 1 violation(s) marked as ignored, likely already known
yaml[commas]: Too many spaces after comma (warning) # ignored
example.yml:3


Passed: 0 failure(s), 1 warning(s) in 1 files processed of 1 encountered. Last profile that met the validation criteria was 'production'.
```
v3: rearrange `example.yml` (now list is first key in example map)
```yaml
example:
  fizz: [buzz,  fizzbuzz]
  foo:   bar
```
Ansible-lint is back to warning about colon:
```
$ ansible-lint example.yml
WARNING  Listing 1 violation(s) marked as ignored, likely already known
yaml[colons]: Too many spaces after colon (warning) # ignored
example.yml:3


Passed: 0 failure(s), 1 warning(s) in 1 files processed of 1 encountered. Last profile that met the validation criteria was 'production'.
```

##### Desired Behavior

Both skip rules should be honored with the example passing lint without any warnings or failures.

##### Actual Behavior

Skip behavior is dependent on the order of directives in `.ansible-lint-ignore` and order of violations encountered in input.

**Comment by santosh7676** [comment]
@pwalkr  This matches the same bug reported in #4730.

Root cause: _rule_is_skipped() stopped on the first non-matching entry in .ansible-lint-ignore instead of checking all rules for the file. With two skip lines for the same file, only the first-loaded rule was fully skipped; the second was treated as a normal ignore (warning with # ignored). That’s why behavior depended on directive order.

Fix: Already merged on main in #5060 (94caed6a — fix: ignore skip lookup across rules). The change replaces return False with continue when the tag doesn’t match the current ignore entry.

This fix is not in 26.4.0 (the version in your report). It landed after the v26.4.0 tag.

**Comment by pwalkr** [comment]
Confirmed: the latest release `26.6.0` honors all skip directives
```console
$ cat example.yml 
example:
  foo:   bar
  fizz: [buzz,  fizzbuzz]

$ cat .ansible-lint-ignore 
example.yml yaml[colons] skip
example.yml yaml[commas] skip

$ ansible-lint example.yml

Passed: 0 failure(s), 0 warning(s) in 1 files processed of 1 encountered. Last profile that met the validation criteria was 'production'.

$ ansible-lint --version
ansible-lint 26.6.0 using ansible-core:2.21.1 ansible-compat:26.6.0 ruamel-yaml:0.19.1 ruamel-yaml-clib:0.2.15
```
